adapters/repos/db/helpers/helpers.go

```diff
@@ -25,6 +29,46 @@ var (
 	DocIDBucket                = []byte("doc_ids")
 )
 
+func MakePropertyPrefix(property string, propIds *tracker.JsonPropertyIdTracker) ([]byte, error) {
+	propid, _ := propIds.GetIdForProperty(string(property))
+	propid_bytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(propid_bytes, propid)
+	return propid_bytes, nil
+}
+
+func MakePropertyKey(propPrefix []byte, key []byte) []byte {
+	if len(propPrefix) == 0 {
+		return nil
+	}
+
+	t := key[:]
+	val := append(t, propPrefix...)
+
+	return val
+}
+
+func MatchesPropertyKeyPrefix(propName []byte, key []byte) bool {
```

Comment:
The key is not the value of the property in the inverted index, it is the the key, which is the property name.  The functions will always be acting on the property name, so I have renamed the variable to "propertyName"

Reply:
actually not true. Property name is used only to create name of directory storing bucket's segments. Which no longer will be the case, as multiple properties will be indexed in single bucket.
Keys in the bucket are values of property, while values in the bucket are bitmaps holding document ids having property with that value/key (exception are properties of type string/text, because keys added to inverted index determined by configured for that property tokenization).
Now keys will be prefixed with propertyId generated by tracker based on property name, but property name still is not part of the key, therefore variable name is misleading)

Reply:
changed, lmkwyt